pkg/sql/unsplit_range_test.go

```diff
@@ -0,0 +1,320 @@
+	}{
+		{
+			name:                   "drop-table-unsplit-sync",
+			query:                  "DROP TABLE t.test1",
+			binaryVersion:          clusterversion.UnsplitRangesInAsyncGCJobs - 1,
+			allKeyCntAfterGC:       0,
+			indexKeyCntAfterGC:     0,
+			hasSplitOnTableAfterGC: false,
+			hasSplitOnKeyAfterGC:   false,
+		},
+		{
+			name:                   "drop-table-unsplit-async",
+			query:                  "DROP TABLE t.test1",
+			binaryVersion:          clusterversion.UnsplitRangesInAsyncGCJobs,
+			allKeyCntAfterGC:       0,
+			indexKeyCntAfterGC:     0,
+			hasSplitOnTableAfterGC: false,
+			hasSplitOnKeyAfterGC:   false,
+		},
+		{
+			name:                   "drop-database-unsplit-sync",
+			query:                  "DROP DATABASE t",
+			binaryVersion:          clusterversion.UnsplitRangesInAsyncGCJobs - 1,
+			allKeyCntAfterGC:       0,
+			indexKeyCntAfterGC:     0,
+			hasSplitOnTableAfterGC: false,
+			hasSplitOnKeyAfterGC:   false,
+		},
+		{
+			name:                   "drop-database-unsplit-async",
+			query:                  "DROP DATABASE t",
+			binaryVersion:          clusterversion.UnsplitRangesInAsyncGCJobs,
+			allKeyCntAfterGC:       0,
+			indexKeyCntAfterGC:     0,
+			hasSplitOnTableAfterGC: false,
+			hasSplitOnKeyAfterGC:   false,
+		},
+		{
+			name:                   "truncate-table-unsplit-sync",
+			query:                  "TRUNCATE TABLE t.test1",
+			binaryVersion:          clusterversion.UnsplitRangesInAsyncGCJobs - 1,
+			allKeyCntAfterGC:       0,
+			indexKeyCntAfterGC:     0,
+			hasSplitOnTableAfterGC: true, // It's true since we copy split points.
+			hasSplitOnKeyAfterGC:   false,
+		},
+		{
+			name:                   "truncate-table-unsplit-async",
+			query:                  "TRUNCATE TABLE t.test1",
+			binaryVersion:          clusterversion.UnsplitRangesInAsyncGCJobs,
+			allKeyCntAfterGC:       0,
+			indexKeyCntAfterGC:     0,
+			hasSplitOnTableAfterGC: true, // It's true since we copy split points.
+			hasSplitOnKeyAfterGC:   false,
+		},
+		{
+			name:                   "truncate-table-unsplit-sync",
+			query:                  "DROP INDEX t.test1@foo",
+			binaryVersion:          clusterversion.UnsplitRangesInAsyncGCJobs - 1,
+			allKeyCntAfterGC:       numRows * 2,
+			indexKeyCntAfterGC:     0,
+			hasSplitOnTableAfterGC: true, // It's true since we only unsplit ranges of index foo
+			hasSplitOnKeyAfterGC:   true,
+		},
+		{
+			name:                   "truncate-table-unsplit-async",
+			query:                  "DROP INDEX t.test1@foo",
+			binaryVersion:          clusterversion.UnsplitRangesInAsyncGCJobs,
+			allKeyCntAfterGC:       numRows * 2,
+			indexKeyCntAfterGC:     0,
+			hasSplitOnTableAfterGC: true, // It's true since we only unsplit ranges of index foo
+			hasSplitOnKeyAfterGC:   true,
+		},
+	}
+
+	for _, testCase := range testCases {
+		t.Run(testCase.name, func(t *testing.T) {
+			defer leaktest.AfterTest(t)()
```

Comment:
just do this one call once at the top of the outermost test (as you have). It'll slow things down otherwise.